tmva/tmva/inc/TMVA/DNN/Minimizers.h

```diff
@@ -148,8 +152,8 @@ class TGradientDescent
     *  determine if the minimization has converged. */
    bool HasConverged();
    /** Increases the minimization step counter by the test error evaluation
-    * period and uses the provided test error value of to determine if
-    * the minimization has converged. */
+    *  period and uses the provided test error value to determine if the
+    *  minimization has converged. */
    bool HasConverged(Scalar_t testError);
```

Comment:
Is this overload actually needed? I find it awkward to have these two public overloads, where that `Has` actually sets something... Then again, the same is true for the other one. Nothing tells me (interface-wise, except the missing `const`-ness...) that `HasConverged` must not be called 5 times in a row - even though it sounds like it's just testing some condition.

Why not split this (in a subsequent PR) into `UpdateConvergeance` and `HasConverged()`, where the latter simply returns `fConvergenceCount >= fConvergenceSteps`?